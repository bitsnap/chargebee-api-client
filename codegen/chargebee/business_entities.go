package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListBusinessEntityTransfers generates chargebee client code to fetch all business entity transfers
//
// API: https://{site}.chargebee.com/api/v2/business_entities/transfers
//
// Documentation: https://apidocs.chargebee.com/docs/api/business_entities?lang=curl#list_the_business_entity_transfers
func GenerateListBusinessEntityTransfers() string {
	return client.GenerateList("BusinessEntityTransfers", "BusinessEntityTransfer", "chargebee.com/api/v2/business_entities/transfers")
}

// GenerateBusinessEntityTransferModel generates chargebee BusinessEntityTransfer domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/business_entities?lang=curl#business_entity_attributes
func GenerateBusinessEntityTransferModel() string {
	return client.GenerateModel("BusinessEntityTransfer", "Business entity transfer attributes", "https://apidocs.chargebee.com/docs/api/business_entity_transfers")
}

// GenerateBusinessEntitiesModel generates chargebee BusinessEntity domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/business_entities?lang=curl#business_entity_attributes
func GenerateBusinessEntitiesModel() string {
	return client.GenerateModel("BusinessEntity", "Business entity attributes", "https://apidocs.chargebee.com/docs/api/business_entities")
}
