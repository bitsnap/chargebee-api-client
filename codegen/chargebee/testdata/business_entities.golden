# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee business entities code generation when generating Chargebee API client should be able to generate business entities model" = '''
(string) (len=406) "type BusinessEntity struct {\n    Id string `json:\"id\" validate:\"required\"`\n    Name string `json:\"name\" validate:\"required\"`\n    Status enums.StatusEnum `json:\"status\" validate:\"required\"`\n    Deleted bool `json:\"deleted\" validate:\"required\"`\n    CreatedAt *time.Time `json:\"created_at\" validate:\"required\"`\n    ResourceVersion int64 `json:\"resource_version\"`\n    UpdatedAt *time.Time `json:\"updated_at\"`\n}"
'''
"Chargebee business entities code generation when generating Chargebee API client should be able to generate business entities transfer model" = '''
(string) (len=38) "type BusinessEntityTransfer struct {\n}"
'''
